### PR TITLE
1608 | Add support for I2C read non stop

### PR DESCRIPTION
**Pull request by felipemoran93**
Resolves https://focusuy.atlassian.net/browse/BMC2-1608
# How to test
## Hardware
You will need a supernova device and an I2C target. You will also need a Saleae Logic to check the I2C signals are correct.
## What to do
Plug the Supernova with the I2C target. Put the Saleae logic to sniff SDA and SCL.
Start the capture in the Saleae sofware and run the following script
```
from supernovacontroller.sequential import SupernovaDevice

def main():

    device = SupernovaDevice()
    print("Opening Supernova host adapter device and getting access to the I2C protocol interface...")
    info = device.open()
    i2c = device.create_interface("i2c")
    print(info)

    print("Initializing the bus...\n")
    (success, _) = i2c.init_bus(3300)
    if not success:
        print("I couldn't initialize the bus. Are you sure there's any target connected?")
        exit(1)

    success, result = i2c.set_parameters(400000)
    if success:
        print(f"Frequency set in: {result} Hz")
    else:
        print(f"Operation failed with error: {result}")
    # Write data
    success, result = i2c.write_non_stop(0x50, [0x00, 0x00], [32 for i in range(1,129)])
    if success:
        print(f'Write operation result: {result} \n')
    else:
        print(f"Operation failed with error: {result}")

    device.close()

if __name__ == "__main__":
    main()
```
## Expected result
You should see in the Saleae software that the I2C transaction started, and a stop condition is not issued afterwards.

### CODE REVIEW

File: supernovacontroller/sequential/i2c.py [hunk 1]
```diff
@@ -252,6 +252,50 @@ def write(self, address, register, data):
 
         return result
 
+    def write_non_stop(self, address, register, data):
+        """
+        Performs a write operation to a specified register on an I2C device, without issuing a stop condition at the end.
+
+        This method sends data to a specified register of a device with a given I2C address, without issuing a stop condition at the end.
+        Before performing the write operation, it checks if the bus voltage is initialized.
```
Comment (joaquin-urrisa) [inline supernovacontroller/sequential/i2c.py:260]: > Before performing the write operation, it checks if the bus voltage is initialized.

Does this happen at the SDK level, should it be added to the code, or is the docstring outdated and should be removed?

**Review by joaquin-urrisa — APPROVED** [review]
LGTM! 
Left a minor comment with a question.

**Comment by Salonso1602** [comment]
<details><summary>Expected SALEAE output:</summary>
<p>

Start of write:
image
End of write:
image
</p>
</details> 

<details><summary>Terminal expected output</summary>
<p>

```
python .\examples\script.py


Opening Supernova host adapter device and getting access to the I2C protocol interface...
{'hw_version': 'C', 'fw_version': '2.4.0', 'serial_number': 'EC68DCA0BE439450A60D391C6FDC1D3D', 'manufacturer': 'Binho LLC', 'product_name': 'Binho Supernova'}
Initializing the bus...

Frequency set in: 400000 Hz
Write operation result: None
```


</p>
</details>

**Review by Salonso1602 — APPROVED** [review]
LGTM

**Pull request merged**